evennia/web/api/filters.py

```diff
@@ -0,0 +1,29 @@
+from django_filters.rest_framework.filterset import FilterSet
+
+from evennia.objects.models import ObjectDB
+from evennia.accounts.models import AccountDB
+from evennia.scripts.models import ScriptDB
+
+
+class TypeclassFilterMixin(object):
```

Comment:
While a mixin works, I find it just makes this code a bit too complex and hard to read in this case - it's just an object that you then have to de-reference just a few lines below. You could just define this list in a global var at the top of this module and use it below.